demo-app/src/main/java/cash/z/ecc/android/sdk/demoapp/util/BenchmarkTrace.kt

```diff
@@ -0,0 +1,106 @@
+package cash.z.ecc.android.sdk.demoapp.util
+
+import android.os.Looper
+import androidx.tracing.Trace
+import cash.z.ecc.android.sdk.ext.BenchmarkingExt
+import cash.z.ecc.android.sdk.internal.twig
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+
+interface BenchmarkTrace {
+    fun checkMainThread() {
+        check(Looper.getMainLooper().thread === Thread.currentThread()) {
+            "Should be called from the main thread, not ${Thread.currentThread()}."
+        }
+    }
+
+    val writeEventMutex: Mutex
+        get() = Mutex()
+
+    interface Event
+}
+
+object SyncBlockchainBenchmarkTrace : BenchmarkTrace {
+    private const val SECTION_BALANCE_SCREEN = "BALANCE_SCREEN" // NON-NLS
+    private const val SECTION_BLOCKCHAIN_SYNC = "BLOCKCHAIN_SYNC"   // NON-NLS
+    private const val SECTION_DOWNLOAD = "DOWNLOAD" // NON-NLS
+    private const val SECTION_VALIDATION = "VALIDATION" // NON-NLS
+    private const val SECTION_SCAN = "SCAN" // NON-NLS
+
+    @Suppress("MagicNumber")
+    suspend fun writeEvent(event: BenchmarkTrace.Event?) {
+        twig("New SyncBlockchain event: $event arrived.")
+        if (!BenchmarkingExt.isBenchmarking()) {
+            return
+        }
+        writeEventMutex.withLock {
```

Comment:
Do we need the mutex if we’re already on the main thread?

Reply:
Nope, we don't. It's a leftover from the previous implementation. Thanks :)